Cython/Utility/ObjectHandling.c

```diff
@@ -1047,10 +1047,20 @@ __Pyx_PEP560_update_bases(PyObject *bases)
     PyObject *base, *meth, *new_base, *result, *new_bases = NULL;
     /*assert(PyTuple_Check(bases));*/
 
+#if CYTHON_ASSUME_SAFE_MACROS
     size_bases = PyTuple_GET_SIZE(bases);
+#else
+    size_bases = PyTuple_Size(bases);
+    if (size_bases < 0) return NULL;
+#endif
     for (i = 0; i < size_bases; i++) {
         // original code in CPython: base  = args[i];
-        base  = PyTuple_GET_ITEM(bases, i);
+#if CYTHON_ASSUME_SAFE_MACROS
+        base = PyTuple_GET_ITEM(bases, i);
+#else
+        base = PyTuple_GetItem(bases, i);
+        if (!base) goto error;
+#endif
```

Comment:
We should incref `base` if we "avoid borrowed refs". We do too many operations below to keep it as a borrowed reference for all that time.